iceoryx_utils/source/posix_wrapper/shared_memory_object/allocator.cpp

```diff
@@ -37,13 +37,9 @@ void* Allocator::allocate(const uint64_t size, const uint64_t alignment) noexcep
 {
     cxx::Expects(size > 0);
 
-    if (m_allocationFinalized)
-    {
-        std::cerr << "allocate() call after finalizeAllocation()! You are not allowed to acquire shared memory chunks "
-                     "anymore!"
-                  << std::endl;
-        std::terminate();
-    }
+    cxx::Expects(
```

Comment:
Here we should also return an `cxx::expected` instead of using `Expects` and `Ensures`.

This error is easily recoverable and the user can handle it if they are informed.

Reply:
In this case I would keep the `Expects` since the user clearly did something wrong. by not providing enough memory to the allocator in the first place

Reply:
I'm torn between here :grimacing: In the end it's about the contract with the user. As this is such a simple class and the user will only use c'tor and the call `allocate()` I'm inclined to just add doxygen hint about the behaviour instead of adding a extra dependency by using `expected.`